### ISSUE TITLE
dosdebug t command traces into int21 calls

### ISSUE THREAD

**Issue by ecm-pushbx** [dosemu2/dosemu2#1171]
**Describe the bug**

When using dosdebug to single-step with the `t` command, singlestepping over an interrupt call doesn't work anymore. It instead steps into the interrupt the same as `ti` is supposed to. This seems to be caused by https://github.com/dosemu2/dosemu2/commit/ce75f0d9c320fd8c6d5a47b5a3ae3f0baaeb7f00 which was a fix for https://github.com/dosemu2/dosemu2/issues/1167

**To Reproduce**

lDebug input:

```
$ dosemu -dumb -K "$PWD" -E "ldebugu.com"
dosemu2 2.0pre8-20200520-1865-g6b138fb25
Configured: 2020-05-20 19:04:08 +0200
Please test against a recent version before reporting bugs and problems.
Get the latest code at http://stsp.github.io/dosemu2
Submit Bugs via https://github.com/stsp/dosemu2/issues
Ask for help in mail list: linux-msdos@vger.kernel.org

FreeDOS kernel - SVN (build 2042 OEM:0xfd) [compiled Sep 22 2017]
Kernel compatibility 7.10 - BORLANDC - FAT32 support

(C) Copyright 1995-2012 Pasquale J. Villani and The FreeDOS Project.
All Rights Reserved. This is free software and comes with ABSOLUTELY NO
WARRANTY; you can redistribute it and/or modify it under the terms of the
GNU General Public License as published by the Free Software Foundation;
either version 2, or (at your option) any later version.
C: HD1, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
D: HD2, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
E: HD3, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
F: HD4, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
EMUFS host file and print access available                                     
dosemu XMS 3.0 & UMB support enabled
dosemu EMS driver rev 0.8 installed.
dosemu CDROM driver installed (V0.2)
Kernel: allocated 30 Diskbuffers = 15960 Bytes in HMA
emufs: redirector enabled
BLASTER=A220 I5 D1 H5 P330 T6
MIDI=SYNTH:2 MAP:E MODE:0
Welcome to dosemu2!
    Build 2.0pre8-20200520-1865-g6b138fb25
About to Execute : ldebugu.com
About to Execute : ldebugu.com
-a 100
1939:0100 mov ah, 09
1939:0102 mov dx, 200
1939:0105 int 21
1939:0107 nop
1939:0108 int3
1939:0109 nop
1939:010A 
-e 200 "Hello world." #13 #10 #36
-g
Hello world.
Unexpected breakpoint interrupt
AX=0924 BX=0000 CX=0000 DX=0200 SP=FFFE BP=0000 SI=0000 DI=0000
DS=1939 ES=1939 SS=1939 CS=1939 IP=0109 NV UP EI PL ZR NA PE NC
1939:0109 90                nop
-g=100
Hello world.
Unexpected breakpoint interrupt
AX=0924 BX=0000 CX=0000 DX=0200 SP=FFFE BP=0000 SI=0000 DI=0000
DS=1939 ES=1939 SS=1939 CS=1939 IP=0109 NV UP EI PL ZR NA PE NC
1939:0109 90                nop
-r ip 100
-g
```

dosdebug input:

```
$ dosdebug

DOSEMU Debugger V0.6 connected
- type ? to get help on commands -
system state: running
AX=0070  BX=98fe  CX=0001  DX=0540  SI=0a3a  DI=0001  SP=09fe  BP=0a2c
DS=00e9  ES=047b  FS=0000  GS=0000  FL=000b0ad6
CS:IP=ff10:01e8       SS:SP=00e9:09fe

ff10:01e8 F4               hlt

dosdebug> stop
system state: stopped
AX=0070  BX=98fe  CX=0001  DX=0540  SI=0a3a  DI=0001  SP=09fe  BP=0a2c
DS=00e9  ES=047b  FS=0000  GS=0000  FL=00030ad6
CS:IP=ff10:01e8       SS:SP=00e9:09fe

ff10:01e8 F4               hlt

dosdebug> bp 1939:100
dosdebug> g
system state: stopped
AX=0924  BX=0000  CX=0000  DX=0200  SI=0000  DI=0000  SP=fffe  BP=0000
DS=1939  ES=1939  FS=0000  GS=0000  FL=000a0246
CS:IP=1939:0100       SS:SP=1939:fffe

1939:0100 B409             mov  ah,09

dosdebug> t
system state: stopped
AX=0924  BX=0000  CX=0000  DX=0200  SI=0000  DI=0000  SP=fffe  BP=0000
DS=1939  ES=1939  FS=0000  GS=0000  FL=000a0346
CS:IP=1939:0102       SS:SP=1939:fffe

1939:0102 BA0002           mov  dx,0200

dosdebug> 
system state: stopped
AX=0924  BX=0000  CX=0000  DX=0200  SI=0000  DI=0000  SP=fffe  BP=0000
DS=1939  ES=1939  FS=0000  GS=0000  FL=000a0346
CS:IP=1939:0105       SS:SP=1939:fffe

1939:0105 CD21             int  21

dosdebug> 
system state: stopped
AX=0924  BX=0000  CX=0000  DX=0200  SI=0000  DI=0000  SP=fff6  BP=0000
DS=1939  ES=1939  FS=0000  GS=0000  FL=000a0346
CS:IP=f000:f6e9       SS:SP=1939:fff6

f000:f6e9 2EFF1EB3F6       call far word cs:[F6B3]

dosdebug> 
```

**Attach the log**

[boot.log](https://github.com/dosemu2/dosemu2/files/4658847/boot.log)

**Additional info**

Running the following shell and kernel:

```
C:\>ver /r
 
FreeCom version 0.84-pre7 - GNUC - XMS_Swap [Dec 29 2019 15:36:33]
DOS version 7.10
FreeDOS kernel - SVN (build 2042 OEM:0xfd) [compiled Sep 22 2017]
C:\>
```

**Comment by stsp** [comment]
But the patch you point to, is only for dpmi!
How could it change that?

**Comment by stsp** [comment]
Not a regression of the patch you point
to, but a regression nevertheless.
Should now be fixed.

**Comment by ecm-pushbx** [comment]
Seems to be fixed in 2.0pre8-1867-g9068c5328, yes.
